snakefmt/formatter.py

```diff
@@ -72,63 +75,112 @@ def read_black_config(self, path: PathLike) -> black.FileMode:
+                )
+                condition = re_match.group(3)
+                if condition != "":
+                    test_substitute = f"{used_keyword}{condition}"
+                else:
+                    test_substitute = f"{used_keyword} a"
+                to_format = (
+                    f"{re_match.group(1)}{test_substitute}"
+                    f"{re_match.group(4)}"
+                    f"{TAB * self.context.cur_indent}pass"
+                )
+                formatted = self.run_black_format_str(to_format)
+                re_rematch = contextual_matcher.match(formatted)
+                if condition != "":
+                    callback_keyword += re_rematch.group(3)
+                formatted = (
+                    f"{re_rematch.group(1)}{callback_keyword}" f"{re_rematch.group(4)}"
+                )
+                formatted_lines = formatted.splitlines(keepends=True)
+                formatted = "".join(formatted_lines[:-1])  # Remove the 'pass' line
+            else:
+                formatted = self.run_black_format_str(self.buffer)
+
+        # Re-add newline removed by black for proper parsing of comments
+        if self.buffer.endswith("\n\n"):
+            formatted += "\n"
+        self.add_newlines(self.target_indent, formatted, final_flush, in_global_context)
         self.buffer = ""
 
-    def process_keyword_context(self):
+    def process_keyword_context(self, in_global_context: bool):
         cur_indent = self.context.cur_indent
-        self.add_newlines(cur_indent, self.context.keyword_name)
+        self.add_newlines(cur_indent, in_global_context=in_global_context)
         formatted = (
-            f"{TAB * cur_indent}{self.context.keyword_name}:{self.context.comment}"
-            + "\n"
+            f"{TAB * cur_indent}{self.context.keyword_name}:{self.context.comment}\n"
         )
         self.result += formatted
 
-    def process_keyword_param(self, param_context):
-        self.add_newlines(param_context.target_indent - 1, param_context.keyword_name)
+    def process_keyword_param(
+        self, param_context: ParameterSyntax, in_global_context: bool
+    ):
+        self.add_newlines(
+            param_context.target_indent - 1, in_global_context=in_global_context
+        )
         in_rule = issubclass(param_context.incident_vocab.__class__, SnakeRule)
         self.result += self.format_params(param_context, in_rule)
 
-    def run_black_format_str(self, string: str, target_indent: int) -> str:
+    def run_black_format_str(self, string: str) -> str:
         try:
             fmted = black.format_str(string, mode=self.black_mode)
         except black.InvalidInput as e:
             raise InvalidPython(
                 f"Got error:\n```\n{str(e)}\n```\n" f"while formatting code with black."
             ) from None
+        return fmted
 
-        # Only indent non-triple-quoted string portions
+    def format_strings(self, string: str, target_indent: int) -> str:
```

Comment:
`align_strings` might a little more descriptive? format is a little vague in the context of this project.

Reply:
makes sense to me, will change